### ISSUE TITLE
strict-ssl / registry settings being ignored but shown correctly by config

### ISSUE THREAD

**Issue by Omortis** [npm/npm#16652]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
`npm config list` returns:

    registry = "http://registry.npmjs.org/"
    strict-ssl = false

`npm config ls -l global` returns:

    ; registry = "https://registry.npmjs.org/" (overridden)
    ; strict-ssl = true (overridden)

but I still get:

    > npm install -g sql-cli
    npm ERR! Windows_NT 6.1.7601
    npm ERR! argv "C:\\Users\\me\\Projects\\localfs\\nodejs\\node.exe" "C:\\Users\\me\\Projects\\localfs\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "sql-cli"
    npm ERR! node v6.10.3
    npm ERR! npm  v3.10.10
    npm ERR! code UNABLE_TO_GET_ISSUER_CERT_LOCALLY

Note that this setup works fine in a linux guest vm here (I can't setup ssl due to corp restrictions...) and of course it all works swimmingly in win10/linux/osx at home.

What am I missing? 
#### How can the CLI team reproduce the problem?
No idea except to override strict-ssl and registry as above _and not have https access_...

### supporting information:

 - `npm -v` prints:
> npm -v
3.10.10

 - `node -v` prints:
> node -v
v6.10.3

 - `npm config get registry` prints:
> npm config get registry
https://registry.npmjs.org/

?????
 - Windows, OS X/macOS, or Linux?:

Windows 7 Enterprise 64-bit

 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
   - [x] behind corporate firewall, no proxy info available in my environment that I can find (but not a windows guru)

Posted on stackoverflow as well but thought I'd follow the error message provided by npm.

Thanks,

John

PS: I have tried to setup node/npm entirely in my own user space in windows.

**Comment by legodude17** [comment]
Try updating npm to the latest version

**Comment by Omortis** [comment]
Huh. Thought npm got bumped up during the node install but I guess they are fully independent.  Broughtmy laptop home and updated npm from 3.10.10 -> 4.6.1. The first time I ran the new version it brought up an issue that may have been effecting 3.10. See below. The 

`                          npm update check failed                            │
│                      Try running with sudo or get access                      │
│                     to the local update config store via                      │
│ sudo chown -R $USER:$(id -gn $USER)  C:\Users\john.bleichert\Projects\.config|
`

I do not have a ~\Projects\.config, but that is the correct directory per my setup. Perhaps that's why npm is getting confused? What is supposed to be in the windows ~/.config? Is that the linux config?

**Comment by legodude17** [comment]
Yes, `npm` is confused because the autoupdater is trying to access that directory but it doesn't exist. Try creating that dir.

**Comment by Omortis** [comment]
I created the .config directory specified in the error message above, and it is empty. Indeed, I would expect to see that dir in linux but not in windows? Still the same behavior - and any `npm config *` commands bring up the square warning box...

**Comment by legodude17** [comment]
Hmm, does `npm config ls` output anything other then the warning box? If so, what is it?

**Comment by Omortis** [comment]
Following the details in the npmrc man page I created a simple .npmrc in the directory I've been trying to test/fix npm in:

```
me@here C:\Users\me\Projects\bootstrap-3.3.7
> cat .npmrc
strict-ssl=false
registry=http://registry.npmjs.org/
```

and in this directory  npm  seems to work correctly now.  I also created `${PREFIX}\etc\npmrc` . 

So npm seems to work ok in a working project directory w/ .npmrc and package.json but elsewhere I still get the error about the missing linux .config dir.

Thanks for sticking with me through this, by the way.

**Comment by legodude17** [comment]
No problem, do you have any `.npmrc` that could be changing the config directory?

**Comment by Omortis** [comment]
@legodude17  in a non-project directory (actually, in $HOME):

```
me@here C:\Users\me\Projects
> npm config ls
; cli configs
metrics-registry = "http://registry.npmjs.org/"
scope = ""
user-agent = "npm/4.6.1 node/v6.10.3 win32 x64"

; project config C:\Users\me\.npmrc
cache = "C:\\Users\\me\\Projects\\cache = \"C:\\Users\\me\\AppData\\Local\\npm-cache\""
prefix = "C:\\Users\\me\\Projects\\npm"
registry = "http://registry.npmjs.org/"
strict-ssl = false

; globalconfig C:\Users\me\Projects\npm\etc\npmrc
registry = "http://registry.npmjs.org/"
strict-ssl = false

; node bin location = C:\Users\me\Projects\localfs\nodejs\node.exe
; cwd = C:\Users\me\Projects
; HOME =  C:\Users\me\Projects
; "npm config ls -l" to show all defaults.
```

In a working project directory:

```
me@here C:\Users\me\Projects\VeRA\reactjs
> npm config ls
; cli configs
metrics-registry = "http://registry.npmjs.org/"
scope = ""
user-agent = "npm/4.6.1 node/v6.10.3 win32 x64"

; project config C:\Users\me\Projects\VeRA\reactjs\.npmrc
registry = "http://registry.npmjs.org/"
strict-ssl = false

; node bin location = C:\Users\me\Projects\localfs\nodejs\node.exe
; cwd = C:\Users\me\Projects\VeRA\reactjs
; HOME =  C:\Users\me\Projects
; "npm config ls -l" to show all defaults.
```

**Comment by legodude17** [comment]
Do commands still give the update checker error message?

**Comment by Omortis** [comment]
Yes, and global installs still don't work. The bootstrap test install worked til the end when I hit some EPERM issues but that's just Win7 being an idiot...

**Comment by Omortis** [comment]
Tried again. Removed all vestiges of node and npm from this laptop. The old 3.x npm that ships with a  vanilla install of node 6.10.3 displayed no errors or warnings. I was able to update npm just fine and everything works, but it still thinks it's on a linux box:

![npm-warning](https://cloud.githubusercontent.com/assets/14285373/26754934/bc2e6d84-4851-11e7-8197-156328ec51a5.png)

This error/warning is not displayed if I run in a terminal started as administrator. Closing the issue, it's just Windows 7 being a putz. Thanks for the feedback!

JB
